roadmap/implementors-guide/guide.md

```diff
@@ -1041,11 +1078,82 @@ struct BlockFinalizationEvent {
 }
 ```
 
+#### Overseer Signal
+
+Signals from the overseer to a subsystem to request change in execution that has to be obeyed by the subsystem.
+
+```rust
+enum OverseerSignal {
+  /// Signal to start work localized to the relay-parent hash.
+  StartWork(Hash),
+  /// Signal to stop (or phase down) work localized to the relay-parent hash.
+  StopWork(Hash),
+}
+```
+
+#### Candidate Selection Subsystem Message
+
+These messages are sent from the overseer to the Candidate Selection subsystem when new parablocks are available for validation.
+
+```rust
+enum CandidateSelectionSubsystemMessage {
+  /// Start or stop work on blocks related to a given relay parent
+  Signal(OverseerSignal),
+  /// A new parachain candidate has arrived from a collator and should be considered for seconding.
+  NewCandidate(PoV, ParachainBlock),
+}
+```
+
+If this subsystem chooses to second a parachain block, it dispatches a `CandidateBackingSubsystemMessage`.
+
+#### Candidate Backing Subsystem Message
+
+```rust
+enum CandidateBackingSubsystemMessage {
+  /// Start or stop work on blocks related to a given relay parent
+  Signal(OverseerSignal),
```

Comment:
to avoid having an instantiation of the overseer signal variant in every subsystem message, I'd suggest omitting it here and having the subsystems accept an `FromOverseer<T> { Signal(OverseerSignal), Communication(T) }`. instead.  This is what is done in the overseer implementation by @montekki as well

Reply:
That's reasonable.

Reply:
Alternately, we could remove an indirection layer by rewriting `OverseerSignal`:

```rust
enum OverseerSignal<T> {
  StartWork(Hash),
  StopWork(Hash),
  Message(T),
}
```